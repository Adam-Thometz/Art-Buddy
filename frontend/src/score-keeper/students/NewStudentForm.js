import React, { useState } from "react";

import { useDispatch } from "react-redux";
import { addStudent } from "../../actions";

const NewStudentForm = () => {
  const dispatch = useDispatch();
  const [studentName, setStudentName] = useState('');

  const handleChange = e => {
    setStudentName(e.target.value);
  };

  const handleSubmit = () => {
    dispatch(addStudent(studentName));
    setStudentName('');
  };

  return (
    <div className="NewStudentForm">
      <label htmlFor="newStudent">Add a student: </label>
      <input
        type="text"
        id="newStudent"
        name="newStudent"
        value={studentName}
        onChange={handleChange}
      />
      <button onClick={handleSubmit}>Add</button>
    </div>
  );
};

export default NewStudentForm;