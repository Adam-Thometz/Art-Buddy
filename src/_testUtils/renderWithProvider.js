import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;