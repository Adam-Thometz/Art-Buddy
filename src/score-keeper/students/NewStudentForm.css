.NewStudentForm {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  height: 75px;
  width: 50%;
  margin: 0 auto;
}