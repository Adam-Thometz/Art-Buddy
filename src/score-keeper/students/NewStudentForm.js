import React, { useState } from "react";

import { useSelector, useDispatch } from "react-redux";
import { addStudent } from "../../_redux/actions/scoreKeeperActions";

import './NewStudentForm.css';

import Dropdown from "../../_components/dropdown/Dropdown";
import Button from "../../_components/button/Button";

import colorOptions from "../_utils/colorOptions";

const NewStudentForm = () => {
  const error = useSelector(state => state.scoreKeeper.error);
  const dispatch = useDispatch();
  const [name, setName] = useState('');
  const [color, setColor] = useState('');

  const handleChangeName = e => {
    setName(e.target.value);
  };

  const handleChangeColor = e => {
    setColor(e.target.id.replace('hex', '#'));
  };

  const handleSubmit = () => {
    dispatch(addStudent({name, color}));
    setName('');
  };

  return (
    <div className="NewStudentForm">
      <div className="NewStudentForm-student-name">
        <label className="NewStudentForm-label" htmlFor="newStudent">ADD A STUDENT: </label>
        <input
          className="NewStudentForm-input"
          type="text"
          id="newStudent"
          name="newStudent"
          value={name}
          onChange={handleChangeName}
        />
        {error ? <p>{error}</p> : null}
      </div>
      <Dropdown
        labelText="PICK A COLOR"
        onClick={handleChangeColor}
        options={colorOptions}
        defaultValue='hexFFFFFF'
      />
      <Button small colorId={3} onClick={handleSubmit}>ADD STUDENT</Button>
    </div>
  );
};

export default NewStudentForm;