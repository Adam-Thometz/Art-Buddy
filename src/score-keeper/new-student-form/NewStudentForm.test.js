import React from "react";

import renderWithProvider from "../../_testUtils/renderWithProvider";

import NewStudentForm from "./NewStudentForm";

describe('New Student Form', () => {
  it('renders without crashing', () => {
    renderWithProvider(<NewStudentForm />)
  });

  it('matches the snapshot', () => {
    const { asFragment } = renderWithProvider(<NewStudentForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});