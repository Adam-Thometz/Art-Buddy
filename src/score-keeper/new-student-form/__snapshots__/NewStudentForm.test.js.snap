// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Student Form matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewStudentForm"
  >
    <div
      class="NewStudentForm-student-name"
    >
      <label
        class="NewStudentForm-label"
        for="newStudent"
      >
        ADD A STUDENT: 
      </label>
      <input
        class="NewStudentForm-input"
        id="newStudent"
        name="newStudent"
        type="text"
        value=""
      />
    </div>
    <div
      class="Dropdown"
    >
      <span
        class="Dropdown-label"
      >
        PICK A COLOR 
        <img
          alt=""
          src="file"
        />
      </span>
    </div>
    <button
      class="Button small"
      style="border-color: #4d4f5e;"
    >
      ADD STUDENT
    </button>
  </div>
</DocumentFragment>
`;
